Monitoring dashboards: Add text filter to the Dashboards dropdown and the variable dropdowns

kyoto opened the pull request:
Jira ticket: https://issues.redhat.com/browse/MON-1521

Also moves `TimespanDropdown` to its own file since it no longer uses `VariableDropdown`.

![screenshot](https://user-images.githubusercontent.com/460802/124763621-26305e00-df6f-11eb-8bf6-9c479be07a63.png)

sichvoge commented:
Does this work for the "Dashboards" drop down as well?

kyoto commented:
@sichvoge I hit a problem with the dashboards dropdown, so decided to leave that for a separate PR.

spadgett commented on frontend/public/components/monitoring/dashboards/index.tsx:
```diff
@@ -105,7 +112,27 @@ const VariableDropdown: React.FC<VariableDropdownProps> = ({
+  };
+
+  const filteredItems = filterText
+    ? _.pickBy(items, (v) => v.toLowerCase().includes(filterText))
```
Worth a comment explaining the `filterText` is lower-cased before it's saved in state.

spadgett reviewed (approved):
/lgtm
/hold for approvals

kyoto commented:
@spadgett I ended up extending and reworking this
- Rebased
- Addressed your feedback comment
- Added the text filter for the Dashboards dropdown too
- Refactored the `isError` case to also use a `Select`

Preeticp commented:
/label docs-approved

lihongyan1 commented:
Filed bug 
https://bugzilla.redhat.com/show_bug.cgi?id=1984365

lihongyan1 commented:
/label qe-approved
The bug is not critical and not very related to the PR

spadgett reviewed (approved):
/lgtm

kyoto commented:
/assign @sferich888
for px- approval

sferich888 commented:
/lables px-approved
/unassign @sferich888

spadgett commented:
/label px-approved
/hold cancel

@kyoto Needs rebase :/

kyoto commented:
@spadgett Rebased

spadgett reviewed (approved):
/lgtm
